Index: ds30 Loader GUI/current/ds30 Loader GUI/clsSettings.cs
===================================================================
--- ds30 Loader GUI/current/ds30 Loader GUI/clsSettings.cs	(revision 218)
+++ ds30 Loader GUI/current/ds30 Loader GUI/clsSettings.cs	(working copy)
@@ -97,5 +97,12 @@
 		        strFilename = value;
 	        }
         }
-    }// Class: settings
+
+		//---------------------------------------------------------------------
+		// Distribution Specific
+		//---------------------------------------------------------------------
+    	public DefaultStartSettings DefaultStartSettings { get; set;}
+    }
+
+// Class: settings
 }
Index: ds30 Loader GUI/current/ds30 Loader GUI/ds30LoaderGUI.csproj
===================================================================
--- ds30 Loader GUI/current/ds30 Loader GUI/ds30LoaderGUI.csproj	(revision 218)
+++ ds30 Loader GUI/current/ds30 Loader GUI/ds30LoaderGUI.csproj	(working copy)
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProductVersion>9.0.21022</ProductVersion>
+    <ProductVersion>9.0.30729</ProductVersion>
     <SchemaVersion>2.0</SchemaVersion>
     <ProjectGuid>{A8CC9936-6178-4566-A68A-2519A61350CC}</ProjectGuid>
     <OutputType>WinExe</OutputType>
@@ -48,7 +48,7 @@
   <ItemGroup>
     <Reference Include="GHelper, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\..\..\C#\GHelper\GHelper\bin\Release\GHelper.dll</HintPath>
+      <HintPath>..\..\..\..\..\..\Projects\BusPirate\svn\ds30Loader\loader-apps\GHelper.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Data" />
@@ -62,6 +62,7 @@
     <Compile Include="clsRecentFile.cs" />
     <Compile Include="clsRecentFiles.cs" />
     <Compile Include="clsSettings.cs" />
+    <Compile Include="DefaultStartSettings.cs" />
     <Compile Include="frmAbout.cs">
       <SubType>Form</SubType>
     </Compile>
Index: ds30 Loader GUI/current/ds30 Loader GUI/frmDS30Loader.cs
===================================================================
--- ds30 Loader GUI/current/ds30 Loader GUI/frmDS30Loader.cs	(revision 218)
+++ ds30 Loader GUI/current/ds30 Loader GUI/frmDS30Loader.cs	(working copy)
@@ -746,7 +746,7 @@
             }
             
             // Tab basic 
-            cboFiles.Text = objSettings.Filename;
+            cboFiles.Text = Path.GetFullPath( objSettings.Filename);
 
 
 			//-----------------------------------------------------------------
@@ -779,9 +779,27 @@
             // If the file isn't a recent file, copy settings from gui
             // ----------------------------------------------------------------
             if ( bExistingRecentFile == false ) {
-                UpdateRecentFile( objRecentFile );
-            
+				if (objSettings.DefaultStartSettings != null)
+				{
+					cboBaudrate.Text = objSettings.DefaultStartSettings.BaudRate;
+					cboFamily.Text = objSettings.DefaultStartSettings.FamilyName;
+					cboDevice.SelectedItem = clsDevices.DeviceGet(cboFamily.Text + objSettings.DefaultStartSettings.DeviceName);
+					chkWriteProgram.Checked = objSettings.DefaultStartSettings.WriteProgram;
+					if (objSettings.DefaultStartSettings.UseCustomBootLoaderPlacement)
+					{
+						txtCustomBlPlacementP.Text = objSettings.DefaultStartSettings.CustomBootLoaderPlacement.ToString();
+						chkCustomBl.Checked = true;						
+					}
 
+					if (objSettings.DefaultStartSettings.ProtectOverwriteBootLoader)
+					{
+						chkAllowBlOverwrite.Enabled = false;
+					}
+				}
+				
+				UpdateRecentFile( objRecentFile );
+				
+
             // ----------------------------------------------------------------
             // It's a recent file, apply settings
             // ----------------------------------------------------------------
@@ -1768,7 +1786,7 @@
 		//---------------------------------------------------------------------
         private void picHexContent_Paint(object sender, PaintEventArgs e)
         {
-            if ( objHex == null ) {
+            if ( objHex == null || objHex.objParsedSettings.device == null) {
                 return;
             }
             //
Index: ds30 Loader GUI/current/ds30 Loader GUI/frmDS30Loader.Designer.cs
===================================================================
--- ds30 Loader GUI/current/ds30 Loader GUI/frmDS30Loader.Designer.cs	(revision 218)
+++ ds30 Loader GUI/current/ds30 Loader GUI/frmDS30Loader.Designer.cs	(working copy)
@@ -448,7 +448,7 @@
             this.label3.Name = "label3";
             this.label3.Size = new System.Drawing.Size(53, 13);
             this.label3.TabIndex = 9;
-            this.label3.Text = "Baudrate:";
+            this.label3.Text = "BaudRate:";
             // 
             // cboCOMPort
             // 
@@ -791,7 +791,7 @@
             this.label10.Name = "label10";
             this.label10.Size = new System.Drawing.Size(53, 13);
             this.label10.TabIndex = 17;
-            this.label10.Text = "Baudrate:";
+            this.label10.Text = "BaudRate:";
             // 
             // cboResetBaudrate
             // 
@@ -1101,7 +1101,7 @@
             this.label9.Name = "label9";
             this.label9.Size = new System.Drawing.Size(53, 13);
             this.label9.TabIndex = 11;
-            this.label9.Text = "Baudrate:";
+            this.label9.Text = "BaudRate:";
             // 
             // toolStrip2
             // 
Index: ds30 Loader GUI/current/ds30LoaderGUI.suo
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream
Index: ds30 Loader/current/ds30 Loader/ds30Loader.csproj
===================================================================
--- ds30 Loader/current/ds30 Loader/ds30Loader.csproj	(revision 218)
+++ ds30 Loader/current/ds30 Loader/ds30Loader.csproj	(working copy)
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProductVersion>9.0.21022</ProductVersion>
+    <ProductVersion>9.0.30729</ProductVersion>
     <SchemaVersion>2.0</SchemaVersion>
     <ProjectGuid>{D7B00D8E-C7D8-4FA4-8440-C6BB109CC9FA}</ProjectGuid>
     <OutputType>Library</OutputType>
@@ -50,7 +50,7 @@
   <ItemGroup>
     <Reference Include="GHelper, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\..\..\C#\GHelper\GHelper\bin\Release\GHelper.dll</HintPath>
+      <HintPath>..\..\..\..\..\..\Projects\BusPirate\svn\ds30Loader\loader-apps\GHelper.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Data" />
